libs/domain/product/availability/availability.component.spec.ts

```diff
@@ -0,0 +1,261 @@
+import { fixture, html } from '@open-wc/testing-helpers';
+import { useComponent } from '@spryker-oryx/core/utilities';
+import { createInjector, destroyInjector } from '@spryker-oryx/di';
+import { ProductService } from '@spryker-oryx/product';
+import { AlertType } from '@spryker-oryx/ui';
+import { LitElement } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+import { of } from 'rxjs';
+import { ProductAvailabilityComponent } from './availability.component';
+import { productAvailabilityComponent } from './availability.def';
+
+class MockProductService implements Partial<ProductService> {
+  get = vi.fn();
+}
+
+@customElement('oryx-swatch')
```

Comment:
Why do you need to mock the component? Just check the property instead of attributes:

```
const getSwatch = () => element.renderRoot.querySelector<SwatchComponent>('oryx-swatch');
 
it('should render oryx-swatch with type success', () => {
        expect(getSwatch()?.type).toBe(AlertType.Success);
});
```

Reply:
because the type is not reflected

Reply:
Reflect works for the attribute, but prop should be presented all time, even in not-defined custom element

Reply:
actually, no need to do it. apparently, when the element is not registered, lit will reflect it as an attribute by default anyway.

Reply:
Yes, in case standard attribute binding ie `attr="value"`, but not dot binding `.attr="value"`

Reply:
actually it does. When there's no registered version of the element, the attributes can be queried. It actually makes sense, as the element can be registered later on, and is this case the property would be lost.